contracts/truefi2/LoanFactory2.sol

```diff
@@ -117,8 +117,11 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
         _;
     }
 
-    modifier onlyFTLA() {
-        require(msg.sender == address(ftlAgency), "LoanFactory: Caller is not the fixed term loan agency");
+    modifier onlyFTLAOrLoan() {
+        require(
+            msg.sender == address(ftlAgency) || isLoanToken[msg.sender],
+            "LoanFactory: Caller is neither the fixed term loan agency not loan token"
```

Comment:
This hunk is obsolete, but just a heads up in case you copy/paste to `onlyTCAOrLoan`.:
```suggestion
            "LoanFactory: Caller is neither the fixed term loan agency nor a loan token"
```